TWW Updates

pranavchary opened the pull request.

pranavchary committed:
TWW Updates
- Decouple Dragonflight calculations and related methods into a separate folder
- Created a new folder with methods and helpers for calculating TWW m+ score
- Added a caching service to potentially fetch and use raider.io data in the future without needing to make multiple requests in a short period

pranavchary committed:
Maintenance Tasks
- Update README
- Modify Gamon's available quotes

pranavchary committed:
Remove DF S3 references

pranavchary committed:
Forgot one lol

KnobleChipmunk commented on README.md:
```diff
@@ -3,16 +3,18 @@
 Just kidding.
 
-Gamon is a Discord bot that can give information about completing Mythic+ dungeons at certain keystone levels and how that might impact a character's Mythic+ rating. This bot is based on [Mythic Rating Helper](https://github.com/Coryrin/mr-helper) and has been modified to work within the Discord server for the World of Warcraft guild The Last Dynasty (US - Thrall), **TLD Reforged**.
+Gamon is a Discord bot that can give information about completing Mythic+ dungeons at certain keystone levels and how that might impact a character's Mythic+ rating. This bot is based on [Mythic Rating Helper](https://github.com/Coryrin/mr-helper) and has been modified to work within the Discord server for the World of Warcraft guild The Last Dynasty (US - Thrall).
```
This could be more specific? "give information about" is a little vague. I realize it's for the readme, but I lean towards concise and abbreviated these days to contend with increasingly shorter attention spans.

pranavchary replied:
Expanded on this more while still attempting to provide a short, easy-to-understand description

KnobleChipmunk commented on README.md:
```diff
@@ -3,16 +3,18 @@
 What Gamon can do for you:
 - Simulate a character running all Mythic+ dungeons at a single keystone level
-- Tell which dungeons a character could run to slightly improve their Mythic+ rating, prioritizing the current week's main affix (Tyrannical or Fortified)
+- Tell which dungeons a character could (theoretically) run and time to slightly improve their Mythic+ rating
```
theoretically seems implicit based on the function of the bot to me

pranavchary replied:
Fair point. What I think in my head should probably not be what's written out and made public haha

KnobleChipmunk commented on commands/cmd.js:
```diff
@@ -0,0 +1,74 @@
+                console.error(e);
+                await interaction.reply({
+                    // content: 'An error occurred while fetching your results. This could be an issue with Radier.io or with Discord itself. Please DM Pran or Tusk what you were trying to do when this happened, or try again in a few moments.',
+                    content: 'An error occurred while fetching your results. This could be an issue with Radier.io or with Discord itself. Please DM Pran (@Pranavius) what you were trying to do when this happened, or try again in a few moments.',
```
typo in raider

KnobleChipmunk commented on constants.js:
```diff
@@ -1,44 +1,31 @@
-};
 const SPEC_SHORTNAME_MAP = {
     Affliction: 'Aff',
     Assassination: 'Sin',
```
I still prefer `Ass` but then I am one haha

pranavchary replied:
I mean, it makes sense given our Guild lore 😂

KnobleChipmunk commented on constants.js:
```diff
@@ -1,44 +1,31 @@
     'For the Horde, my friend.',
-    'I may be bruised and beaten, but the hatred boils inside me.'
+    'I may be bruised and beaten, but the hatred boils inside me.',
+    'I need a new axe... My current one, it has cleaved too many skulls.'
```
Would be cool to find a way to pull these in dynamically from somewhere rather than have to define them (down the road)

pranavchary replied:
Good thinking. We can see if there's a resource that exists for quotes and such

KnobleChipmunk commented on df-presquish/commands.js:
```diff
@@ -394,7 +390,7 @@ const requestRaiderIoData = async (qp) => {
                 name: qp.name.toLowerCase(),
                 region: 'us',
-                fields: 'gear,mythic_plus_best_runs,mythic_plus_alternate_runs,mythic_plus_scores_by_season:current'
+                fields: 'gear,mythic_plus_best_runs,mythic_plus_alternate_runs,mythic_plus_scores_by_season:season-df-3'
```
umm df s3?

pranavchary replied:
Yeah, the file here is specifically for the Dragonflight Mythic+ scores before the level squish in Season 4 and isn't being used by the bot. I didn't want to completely delete so it could still be used as a reference for TWW updates. Just changed the data retrieved to clearly differentiate it from the current version.

KnobleChipmunk reviewed (dismissed):
I had a nice comment here and github decided to erase it. Looks good. Love how easy the code is to read even though I still don't understand some of it. Added comments for typos and such. No re-approval needed. Thanks man.

pranavchary committed:
Address PR review comments
- Update README
- Fix typo in "Raider.io"
- Ass >:)

pranavchary committed:
Restrict max key level
- Add a limit so simulated or calculated key scores dont go higher than the current maximum level we have base score for
- Updated README
- Butts...

pranavchary committed:
Make Butts Public Again

pranavchary committed:
Merge branch 'dev' into update/new-mplus-criteria

pranavchary committed:
Minor Updates
- Replaced ephemeral property with appropriate MessageFlag (discordjs update)
- Show correct data for 0 rating characters
